### PR TITLE
WEBC-558

### PR DESCRIPTION
**Pull request by awesselburg**

### CODE REVIEW

File: src/app/code/community/Shopgate/Cloudapi/Block/Checkout/Onepage/Success.php [hunk 1]
```diff
@@ -73,37 +73,42 @@ public function getJsonOrderData()
...
+            $price->setData('withTax', $this->formatPrice($item->getPriceInclTax()));
+            $price->setData('net', $this->formatPrice($item->getPrice()));
+            $product->addPrice($price);
+            $purchase->addProduct($product);
         }
 
-        return $logPurchase->getJsonData();
+        /** @var Shopgate_Cloudapi_Model_Frontend_Checkout_Onepage_Purchase_Total $total */
+        $total = Mage::getModel('shopgate_cloudapi/frontend_checkout_onepage_purchase_total');
+
```
Comment (andre-kraus) [inline src/app/code/community/Shopgate/Cloudapi/Block/Checkout/Onepage/Success.php]: please remove the blank line

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Checkout/Onepage/Purchase/Product/Price.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @author    Shopgate Inc, 804 Congress Ave, Austin, Texas 78701 <interfaces@shopgate.com>
+ * @copyright Shopgate Inc
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ */
+
+class Shopgate_Cloudapi_Model_Frontend_Checkout_Onepage_Purchase_Product_Price extends Varien_Object
```
Comment (andre-kraus) [inline src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Checkout/Onepage/Purchase/Product/Price.php]: Please add PHPDoc for every method that should be used (same as you did for product)

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Checkout/Onepage/Purchase/Total.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @author    Shopgate Inc, 804 Congress Ave, Austin, Texas 78701 <interfaces@shopgate.com>
+ * @copyright Shopgate Inc
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ */
+
+class Shopgate_Cloudapi_Model_Frontend_Checkout_Onepage_Purchase_Total extends Varien_Object
```
Comment (andre-kraus) [inline src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Checkout/Onepage/Purchase/Total.php]: Please add PHPDoc for every method that should be used (same as you did for product)

File: src/app/code/community/Shopgate/Cloudapi/Block/Checkout/Onepage/Success.php [hunk 4]
```diff
@@ -73,37 +73,42 @@ public function getJsonOrderData()
...
+        $total->setData('type', 'shipping');
+        $total->setData('amount', $this->formatPrice($order->getShippingInclTax()));
+        $purchase->addTotal($total);
```
Comment (andre-kraus) [inline src/app/code/community/Shopgate/Cloudapi/Block/Checkout/Onepage/Success.php]: Do not use use setDate, use real method + document them first using PHPDoc in the total class

File: src/app/code/community/Shopgate/Cloudapi/Block/Checkout/Onepage/Success.php [hunk 5]
```diff
@@ -73,37 +73,42 @@ public function getJsonOrderData()
...
-                    ->setShippingNet(
-                        $this->formatPrice($order->getShippingAmount() - $order->getShippingTaxAmount())
-                    )
-                    ->setCurrency($order->getStoreCurrencyCode())
-                    ->setSuccess(true)
-                    ->setBlacklist(false)
-                    ->setTrackers(array());
+        /** @var Shopgate_Cloudapi_Model_Frontend_Checkout_Onepage_Purchase $purchase */
+        $purchase = Mage::getModel('shopgate_cloudapi/frontend_checkout_onepage_purchase');
+        $purchase->setNumber($order->getId());
```
Comment (andre-kraus) [inline src/app/code/community/Shopgate/Cloudapi/Block/Checkout/Onepage/Success.php]: This should be the increment_id, according to the example at https://wiki.shopgate.guru/display/DEV/Web+Checkout+and+Web+Register

File: src/app/code/community/Shopgate/Cloudapi/Block/Checkout/Onepage/Success.php [hunk 6]
```diff
@@ -73,37 +73,42 @@ public function getJsonOrderData()
...
-                                   ->setType('product')
-                                   ->setName($item->getName())
-                                   ->setPriceNet($this->formatPrice($item->getPriceInclTax()))
-                                   ->setPriceGross($this->formatPrice($item->getPrice()))
-                                   ->setQuantity($item->getQtyOrdered());
-            $logPurchase->addItem($logPurchaseItem);
+            /** @var Shopgate_Cloudapi_Model_Frontend_Checkout_Onepage_Purchase_Product $product */
+            $product = Mage::getModel('shopgate_cloudapi/frontend_checkout_onepage_purchase_product');
+            $product->setId($item->getId());
+            $product->setQuantity($item->getQtyOrdered());
```
Comment (andre-kraus) [inline src/app/code/community/Shopgate/Cloudapi/Block/Checkout/Onepage/Success.php]: According to the example you posted in the Ticket, this is returning a float -> should be an int.

File: src/app/code/community/Shopgate/Cloudapi/Block/Checkout/Onepage/Success.php [hunk 7]
```diff
@@ -73,37 +73,42 @@ public function getJsonOrderData()
...
-                                   ->setId($item->getId())
-                                   ->setType('product')
-                                   ->setName($item->getName())
-                                   ->setPriceNet($this->formatPrice($item->getPriceInclTax()))
-                                   ->setPriceGross($this->formatPrice($item->getPrice()))
-                                   ->setQuantity($item->getQtyOrdered());
-            $logPurchase->addItem($logPurchaseItem);
+            /** @var Shopgate_Cloudapi_Model_Frontend_Checkout_Onepage_Purchase_Product $product */
+            $product = Mage::getModel('shopgate_cloudapi/frontend_checkout_onepage_purchase_product');
+            $product->setId($item->getId());
```
Comment (andre-kraus) [inline src/app/code/community/Shopgate/Cloudapi/Block/Checkout/Onepage/Success.php]: The example at https://wiki.shopgate.guru/display/DEV/Web+Checkout+and+Web+Register shows "SG40" as id, not sure if this should be the uid or sku from the XML file. Could you please verify?

File: src/app/code/community/Shopgate/Cloudapi/Block/Checkout/Onepage/Success.php (hunk 5 above)
Comment (awesselburg) [reply src/app/code/community/Shopgate/Cloudapi/Block/Checkout/Onepage/Success.php]: done

File: src/app/code/community/Shopgate/Cloudapi/Block/Checkout/Onepage/Success.php (hunk 7 above)
Comment (awesselburg) [reply src/app/code/community/Shopgate/Cloudapi/Block/Checkout/Onepage/Success.php]: done

File: src/app/code/community/Shopgate/Cloudapi/Block/Checkout/Onepage/Success.php (hunk 1 above)
Comment (awesselburg) [reply src/app/code/community/Shopgate/Cloudapi/Block/Checkout/Onepage/Success.php]: ... done

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Checkout/Onepage/Purchase/Product/Price.php (hunk 2 above)
Comment (awesselburg) [reply src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Checkout/Onepage/Purchase/Product/Price.php]: done

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Checkout/Onepage/Purchase/Total.php (hunk 3 above)
Comment (awesselburg) [reply src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Checkout/Onepage/Purchase/Total.php]: done

File: src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Checkout/Onepage/Purchase.php [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+ * limitations under the License.
+ *
+ * @author    Shopgate Inc, 804 Congress Ave, Austin, Texas 78701 <interfaces@shopgate.com>
+ * @copyright Shopgate Inc
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ */
+
+/**
+ * @method $this setNumber(string $number)
+ * @method string getNUmber()
```
Comment (andre-kraus) [inline src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Checkout/Onepage/Purchase.php]: Typo -> getNumber not getNUmber

**Pull request merged**